dibi/drivers/DibiOracleDriver.php

```diff
@@ -97,9 +151,50 @@ public function disconnect()
 	public function query($sql)
 	{
 		$res = oci_parse($this->connection, $sql);
+		if ($this->binds) {
+			set_error_handler(function ($errno, $errstr, $errfile, $errline, $errcontext) use ($sql) {
```

Comment:
This error handler is not restored when exception is throw. Is possible to detect error in different way?